src/App.tsx

```diff
@@ -1,35 +1,46 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useEffect, useState } from "react";
+import "./styles/App.css";
+import SurvivorSelection from "./components/SurvivorSelection";
+import urls from "./utilities/urls"
+import { Survivor, UserSelection } from "./utilities/types";
+import Instructions from "./components/Instructions";
+
 
 function App() {
-  const [count, setCount] = useState(0)
+    const [survivorList, setsurvivorList] = useState<Survivor[]>([]); //*to store list of survivors from API pull
+    const [selectedSurvivor, setSelectedSurvivor] = useState<Survivor>({} as Survivor); //*to store survivor selected by user
+    const [userSelection, setUserSelection] = useState<UserSelection>({} as UserSelection)
+
+    useEffect(() => {
+      const fetchSurvivors = async () => {
+        try {
+          const response = await fetch(urls.survivorsURL);
+          if (response.ok) {
+            const data = await response.json();
+            const survivorData: Survivor[] = data.map( (survivor: any) => ({
```

Comment:
nice typing but, can be better by doing the below

```suggestion
            const survivorData: Survivor[] = data.map( (survivor: Survivor) => ({
```

Reply:
avoid using `any` -- you 99.99999% never need to add `any` in TypeScript

Reply:
![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/141854177/d6a3770a-f7ec-4f17-82e2-c91144196d63)
changing type from 'any' to survivor brings up this error. App still functions fine with the mentioned change

Reply:
oh lol, looks like that's a real error

![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/16410698/eb3f11bf-10f2-4e5a-8202-4c0a213867c1)

---

the only reason adding strong typing to the `survivor` element causes this error is because `survivor._id` should be `survivor.id` if we are following the Survivor interface defined in types.tsx